### PR TITLE
 test(compiler): allow asserting matching identifier names

### PR DESCRIPTION
**Pull request by vicb**
We will need to assert identifier names for i18n (`/MSG_[A-Z_]+/`)

### CODE REVIEW

File: packages/compiler/test/render3/mock_compile.ts [hunk 1]
```diff
@@ -20,83 +20,94 @@ import {compilePipe} from '../../src/render3/r3_pipe_compiler';
...
     } else if (token === ELLIPSIS) {
       pieces.push(SKIP);
     } else {
       pieces.push(token);
     }
   }
+
+  if (pieces.length === 0 || TOKEN.lastIndex !== 0) {
+    const from = TOKEN.lastIndex;
+    const to = from + 30;
```
Comment (ocombe) [inline packages/compiler/test/render3/mock_compile.ts]: why "+30"? use a const for magic numbers ;)

File: packages/compiler/test/render3/mock_compile.ts [hunk 2]
```diff
@@ -20,83 +20,94 @@ import {compilePipe} from '../../src/render3/r3_pipe_compiler';
...
+        }
+      }
+    }
   }
 }
 
 const IDENT_LIKE = /^[a-z][A-Z]/;
-const SPECIAL_RE_CHAR = /\/|\(|\)|\||\*|\+|\[|\]|\{|\}|\$/g;
-function r(...pieces: (string | RegExp)[]): RegExp {
+function r(pieces: (string | RegExp)[]): {regexp: RegExp, groups: Map<string, number>} {
```
Comment (ocombe) [inline packages/compiler/test/render3/mock_compile.ts]: What is `r` doing? Any reason for a single letter name? Or maybe add a comment that explains what the function does

**Review by chuckjaz — APPROVED** [review]

**Comment by vicb** [comment]
Need to be rebased once master is green (the failures are not related to the changes in this PR).

**Comment by vicb** [comment]
http://test/OCL:189801199:BASE:189801247:1521578595666:988ec562

**Comment by IgorMinar** [comment]
master is green but affected by sauce-labs flakes :-(
I'm rerunning the tests now

**Pull request closed without merging**